src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1005,8 +982,42 @@ private AdjustmentRule GetPreviousAdjustmentRule(AdjustmentRule rule)
             }
             return cachedData.m_readOnlySystemTimeZones;
         }
-#endif // FEATURE_WIN32_REGISTRY
 
+        private static void PopulateAllSystemTimeZones(CachedData cachedData)
+        {
+#if FEATURE_WIN32_REGISTRY
+            PermissionSet permSet = new PermissionSet(PermissionState.None);
+            permSet.AddPermission(new RegistryPermission(RegistryPermissionAccess.Read, c_timeZonesRegistryHivePermissionList));
+            permSet.Assert();
+            
+            using (RegistryKey reg = Registry.LocalMachine.OpenSubKey(
+                c_timeZonesRegistryHive,
+#if FEATURE_MACL
+                RegistryKeyPermissionCheck.Default,
+                System.Security.AccessControl.RegistryRights.ReadKey
+#else
+                false
+#endif
+                )) {
+
+                if (reg != null) {
+                    foreach (string keyName in reg.GetSubKeyNames()) {
+                        TimeZoneInfo value;
+                        Exception ex;
+                        TryGetTimeZone(keyName, false, out value, out ex, cachedData);  // populate the cache
+                    }
+                }
+            }
+#else // FEATURE_WIN32_REGISTRY
```

Comment:
The conditionals in this file are a little messy, see https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/TimeZoneInfo.cs#L2071.

I can change both of these locations as requested.  But that means if someone tries compiling mscorlib without one of these conditionals set to true, the compile is going to fail.  I'm assuming that is exactly what we want... correct?